pennylane/queuing.py

```diff
@@ -173,6 +173,22 @@ class QueuingError(Exception):
     """Exception that is raised when there is a queuing error"""
 
 
+class WrappedObj:
+    """Wraps an object to make its has dependent on its identity"""
+
+    def __init__(self, obj):
+        self.obj = obj
+
+    def __hash__(self):
+        return id(self.obj)
+
+    def __eq__(self, other):
+        return id(self.obj) == id(other.obj)
+
+    def __repr__(self):
+        return self.obj.__repr__()
```

Comment:
Probably best to differentiate between and object and its wrapped version in the repr. Otherwise, things could get hard to debug:
```suggestion
        return f"Wrapped( {self.obj.__repr__()})"
```

Reply:
Good idea